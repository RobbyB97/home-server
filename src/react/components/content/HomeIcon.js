import React from 'react'


export default class HomeIcon extends React.Component {

    render() {
        return (
            <div className="homeIcon">
                <a className="homeIcon__link" title={this.props.name} href={this.props.link}>
                    <img className="homeIcon__icon" src={this.props.icon} />
                </a>
            </div>
        )
    }
}