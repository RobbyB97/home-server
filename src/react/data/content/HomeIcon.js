export const homeIcon_GitHub = {
    link: "https://www.github.com/robbyb97",
    name: "GitHub",
    icon: "/dist/images/icons/github.svg"
}

export const homeIcon_LinkedIn = {
    link: "https://www.linkedin.com/in/robbyb97",
    name: "LinkedIn",
    icon: "/dist/images/icons/linkedin.png"
}