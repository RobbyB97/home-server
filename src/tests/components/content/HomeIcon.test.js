import React from 'react'
import {shallow} from 'enzyme'

import HomeIcon from '../../../react/components/content/HomeIcon'


test('Render HomeIcon without props', () => {
    const wrapper = shallow(<HomeIcon />)

    expect(wrapper).toMatchSnapshot()
})