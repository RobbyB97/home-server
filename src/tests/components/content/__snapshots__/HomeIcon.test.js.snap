// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HomeIcon without props 1`] = `
<div
  className="homeIcon"
>
  <a
    className="homeIcon__link"
  >
    <img
      className="homeIcon__icon"
    />
  </a>
</div>
`;
